services/productService.js

```diff
@@ -0,0 +1,19 @@
+const { productDao,reviewDao } = require("../models");
+
+const detailProduct = async ( productId ) => {
+    const product = await productDao.getProductById( productId );
+    if ( !product ) {
+        const err = new Error("INVALID_PRODUCT");
+        err.statusCode = 406;
+        throw err;
+    }
+    const detailProduct = await productDao.getProductDetail( productId );
```

Comment:
productDao의 `getProductById` 함수와 `getProductDetail` 함수가 분리되어 있을 필요가 있는지 고민해보면 좋을 것 같습니다.
두 함수 모두 productId를 전달받아서 product 정보를 반환하는 역할을 하고 있기 때문에, 하나로 합쳐서 사용할 수 있겠습니다.

```suggestion
    const product = await productDao.getProductById( productId );
    if ( !product ) {
        const err = new Error("INVALID_PRODUCT");
        err.statusCode = 406;
        throw err;
    }
```